diff -up gtkhtml-3.32.2/gtkhtml/htmlobject.c.par-style-draw gtkhtml-3.32.2/gtkhtml/htmlobject.c
--- gtkhtml-3.32.2/gtkhtml/htmlobject.c.par-style-draw	2010-09-13 14:16:11.000000000 +0200
+++ gtkhtml-3.32.2/gtkhtml/htmlobject.c	2013-07-03 16:39:15.179123742 +0200
@@ -946,6 +946,9 @@ void
 html_object_set_parent (HTMLObject *o, HTMLObject *parent)
 {
 	o->parent = parent;
+
+	/* parent change requires recalc of everything */
+	o->change = HTML_CHANGE_ALL;
 }
 
 static void
