diff -up gtkhtml-3.32.2/gtkhtml/htmlframe.c.iframe-print-height gtkhtml-3.32.2/gtkhtml/htmlframe.c
--- gtkhtml-3.32.2/gtkhtml/htmlframe.c.iframe-print-height	2010-09-13 14:16:11.000000000 +0200
+++ gtkhtml-3.32.2/gtkhtml/htmlframe.c	2017-01-03 12:22:55.438894273 +0100
@@ -241,6 +241,10 @@ html_frame_real_calc_size (HTMLObject *o
 		o->width = e->width;
 		o->ascent = e->height;
 		o->descent = 0;
+
+		if (G_OBJECT_TYPE (painter) == HTML_TYPE_PRINTER) {
+			o->ascent += html_painter_get_pixel_size (painter) * (html_engine_get_top_border (e) + html_engine_get_bottom_border (e));
+		}
 	} else
 		return (* HTML_OBJECT_CLASS (parent_class)->calc_size) (o, painter, changed_objs);
 
diff -up gtkhtml-3.32.2/gtkhtml/htmliframe.c.iframe-print-height gtkhtml-3.32.2/gtkhtml/htmliframe.c
--- gtkhtml-3.32.2/gtkhtml/htmliframe.c.iframe-print-height	2010-09-13 14:16:11.000000000 +0200
+++ gtkhtml-3.32.2/gtkhtml/htmliframe.c	2017-01-03 12:22:55.438894273 +0100
@@ -352,6 +352,10 @@ html_iframe_real_calc_size (HTMLObject *
 		o->width = e->width;
 		o->ascent = e->height;
 		o->descent = 0;
+
+		if (G_OBJECT_TYPE (painter) == HTML_TYPE_PRINTER) {
+			o->ascent += html_painter_get_pixel_size (painter) * (html_engine_get_top_border (e) + html_engine_get_bottom_border (e));
+		}
 	} else
 		return (* HTML_OBJECT_CLASS (parent_class)->calc_size) (o, painter, changed_objs);
 
