diff -up gtkhtml-3.32.2/gtkhtml/htmlinterval.c.covscan-issues gtkhtml-3.32.2/gtkhtml/htmlinterval.c
--- gtkhtml-3.32.2/gtkhtml/htmlinterval.c.covscan-issues	2010-09-13 14:16:11.000000000 +0200
+++ gtkhtml-3.32.2/gtkhtml/htmlinterval.c	2013-07-03 16:11:09.044124226 +0200
@@ -418,7 +418,7 @@ html_interval_validate (HTMLInterval *i)
 gboolean
 html_point_eq (const HTMLPoint *a, const HTMLPoint *b)
 {
-	return a->object == b->object && a->offset == b->offset;
+	return a && b && a->object == b->object && a->offset == b->offset;
 }
 
 gboolean
