diff -up gtkhtml-3.32.2/gtkhtml/htmlcluev.c.cursor-pos-after-paste gtkhtml-3.32.2/gtkhtml/htmlcluev.c
--- gtkhtml-3.32.2/gtkhtml/htmlcluev.c.cursor-pos-after-paste	2010-09-13 14:16:11.000000000 +0200
+++ gtkhtml-3.32.2/gtkhtml/htmlcluev.c	2013-07-03 16:15:42.649124100 +0200
@@ -892,6 +892,20 @@ html_cluev_real_get_direction (HTMLObjec
 	return HTML_CLUEV (o)->dir;
 }
 
+static guint
+get_recursive_length (HTMLObject *self)
+{
+	HTMLClueV *cluev = HTML_CLUEV (self);
+	guint len;
+
+	len = (* HTML_OBJECT_CLASS (parent_class)->get_recursive_length) (self);
+
+	if (cluev->display == DISPLAY_BLOCK)
+		len++;
+
+	return len;
+}
+
 static void
 html_cluev_destroy (HTMLObject *self)
 {
@@ -941,6 +955,7 @@ html_cluev_class_init (HTMLClueVClass *k
 	object_class->get_left_margin = get_left_margin;
 	object_class->get_right_margin = get_right_margin;
 	object_class->get_direction = html_cluev_real_get_direction;
+	object_class->get_recursive_length = get_recursive_length;
 	object_class->destroy = html_cluev_destroy;
 
 	clue_class->get_left_clear = get_left_clear;
@@ -983,6 +998,7 @@ html_cluev_init (HTMLClueV *cluev,
 	cluev->border_width = 0;
 	cluev->border_color = NULL;
 	cluev->background_color = NULL;
+	cluev->display = DISPLAY_INLINE;
 }
 
 HTMLObject *
@@ -1017,6 +1033,7 @@ html_cluev_set_style (HTMLClueV *cluev, 
 		cluev->background_color = style->bg_color;
 		if (cluev->background_color)
 			html_color_ref (cluev->background_color);
+		cluev->display = style->display;
 	} else {
 		if (cluev->border_color)
 			html_color_unref (cluev->border_color);
@@ -1028,5 +1045,6 @@ html_cluev_set_style (HTMLClueV *cluev, 
 		cluev->border_width = 0;
 		cluev->border_color = NULL;
 		cluev->background_color = NULL;
+		cluev->display = DISPLAY_INLINE;
 	}
 }
diff -up gtkhtml-3.32.2/gtkhtml/htmlcluev.h.cursor-pos-after-paste gtkhtml-3.32.2/gtkhtml/htmlcluev.h
--- gtkhtml-3.32.2/gtkhtml/htmlcluev.h.cursor-pos-after-paste	2010-09-13 14:16:11.000000000 +0200
+++ gtkhtml-3.32.2/gtkhtml/htmlcluev.h	2013-07-03 16:15:42.650124100 +0200
@@ -44,6 +44,7 @@ struct _HTMLClueV {
 	HTMLColor *background_color;
 
 	HTMLDirection dir;
+	HTMLDisplayType display;
 };
 
 struct _HTMLClueVClass {
